website/server/models/group.js

```diff
@@ -315,11 +315,76 @@ schema.statics.getGroups = async function getGroups (options = {}) {
   return groupsArray;
 };
 
+function translateSystemMessages (group, user) {
+  let foundText = '';
+  for (let i = 0; i < group.chat.length; i++) {
+    if (!_.isEmpty(group.chat[i].info)) {
+      switch (group.chat[i].info.type) {
+        case 'quest_start':
+          group.chat[i].text = `\`${shared.i18n.t('chatQuestStarted', {questName: questScrolls[group.chat[i].info.quest].text(user.preferences.language)}, user.preferences.language)}\``;
```

Comment:
can you cache the user.preferences.language value in a variable at the beginning of this function? it'll make the code more readable. Like `let lang = user.preferences.lang`

Reply:
Also group.chat[i] at the beginning of the for